protozfitsreader/tests/test_read_raw_events.py

```diff
@@ -0,0 +1,292 @@
+
+    for actual, expected in zip(pixel_flags, expected_pixel_flags):
+        assert (actual == expected).all()
+
+
+def test_local_time():
+    from protozfitsreader import ZFile
+    local_time = [
+        event.local_time
+        for event in ZFile(example_file_path)
+    ]
+    assert local_time == EXPECTED_LOCAL_TIME
+
+
+def test_gps_time():
+    from protozfitsreader import ZFile
+    gps_time = [
+        event.central_event_gps_time
+        for event in ZFile(example_file_path)
+    ]
+    assert gps_time == EXPECTED_GPS_TIME
+
+
+def test_camera_event_type():
+    from protozfitsreader import ZFile
+    camera_event_type = [
+        event.camera_event_type
+        for event in ZFile(example_file_path)
+    ]
+    assert camera_event_type == [1, 1, 1, 1, 1, 8, 1, 1, 1, 1]
+
+
+def test_array_event_type():
+    from protozfitsreader import ZFile
+    array_event_type = [
+        event.array_event_type
+        for event in ZFile(example_file_path)
+    ]
+    assert array_event_type == [0] * EVENTS_IN_EXAMPLE_FILE
+
+
+def test_num_samples():
+    from protozfitsreader import ZFile
+    num_samples = [
+        event.num_samples
+        for event in ZFile(example_file_path)
+    ]
+    assert num_samples == [50] * EVENTS_IN_EXAMPLE_FILE
+
+
+def test_adc_samples():
+    from protozfitsreader import ZFile
+    adc_samples = [
+        event.adc_samples
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in adc_samples:
+        assert actual.dtype == np.int16
+        assert actual.shape == (1296, 50)
+
+    adc_samples = np.array(adc_samples)
+
+    # these are 12 bit ADC values, so the range must
+    # can at least be asserted
+    assert adc_samples.min() == 0
+    assert adc_samples.max() == (2**12) - 1
+
+
+def test_trigger_input_traces():
+    from protozfitsreader import ZFile
+    trigger_input_traces = [
+        event.trigger_input_traces
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in trigger_input_traces:
+        assert actual.dtype == np.uint8
+        assert actual.shape == (432, 50)
+
+
+def test_trigger_output_patch7():
+    from protozfitsreader import ZFile
+    trigger_output_patch7 = [
+        event.trigger_output_patch7
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in trigger_output_patch7:
+        assert actual.dtype == np.uint8
+        assert actual.shape == (432, 50)
+
+
+def test_trigger_output_patch19():
+    from protozfitsreader import ZFile
+    trigger_output_patch19 = [
+        event.trigger_output_patch19
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in trigger_output_patch19:
+        assert actual.dtype == np.uint8
+        assert actual.shape == (432, 50)
+
+
+def test_baseline():
+    from protozfitsreader import ZFile
+    baseline = [
+        event.baseline
+        for event in ZFile(example_file_path)
+    ]
+
+    for actual in baseline:
+        assert actual.dtype == np.int16
```

Comment:
uint16 not int16

Reply:
Are you sure about this? Tests do not pass anymore when I put uint16 here.

Reply:
can be ignored, c.f. comment below